frontend/src/screens/apps/ShowApp.tsx

```diff
@@ -95,7 +103,15 @@ function AppInternal({ app, refetchApp, capabilities }: AppInternalProps) {
 
   const handleSave = async () => {
     try {
+      if (
+        isEditingName &&
+        apps?.some((existingApp) => existingApp.name === name)
+      ) {
+        throw new Error("A connection with the same name already exists.");
```

Comment:
this also happens if you click edit and then save without changing, and there is no cancel button

Reply:
Maybe just check if the app id is different too?

Reply:
Oh, lol true. Yup, will add app id check 👍